<?xml version="1.0"?>
<launch>
    <arg name="rviz" default="false" /> 
    
    <!-- 使用优化的定位节点 -->
    <node pkg="lio_localization" type="global_localization_old.py" name="fast_localization" output="screen">
        <!-- 优化参数 -->
        <param name="MAP_VOXEL_SIZE" value="0.05" />
        <param name="SCAN_VOXEL_SIZE" value="0.01" />
        <param name="FREQ_LOCALIZATION" value="2" />  <!-- 提高频率 -->
        <param name="LOCALIZATION_TH_init" value="0.5" />  <!-- 降低初始阈值 -->
        <param name="LOCALIZATION_TH" value="0.3" />  <!-- 降低阈值 -->
        <param name="FOV" value="6.28" />  <!-- 全视野 -->
        <param name="FOV_FAR" value="100" />  <!-- 更远的视野 -->
    </node>
    
    <!-- transform fusion -->
    <node pkg="lio_localization" type="transform_fusion.py" name="transform_fusion" output="screen" />
    
    <!-- rviz display -->
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" 
              args="-d $(find lio_localization)/rviz/localization.rviz" />
    </group>
</launch>